src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  const adder = (a) => {
+    if (a === undefined) {
+      return 0;
+    }
+
+    return b => {
+      if (b === undefined) {
+        return a;
+      }
+
+      return adder(a + b);
+    };
```

Comment:
no need to create another function, only one function (adder) is allowed

create a variable sum inside makeInfinityAdder and add to it each number passed in the arguments until it encounters a call without parameters